### ISSUE TITLE
Issues with poetry installation

### ISSUE THREAD

**Issue by Hiyorimi** [sky-ring/rift#8]
Had trouble working from inside poetry environment:

```bash
➜  rift-test ~/.pyenv/versions/3.10.7/bin/poetry add rift-framework                                          14.09.22 13:43
The following packages are already present in the pyproject.toml and will be skipped:

  • rift-framework

If you want to update it to the latest compatible version, you can use `poetry update package`.
If you prefer to upgrade it to the latest available version, you can use `poetry add package@latest`.

Nothing to add.
➜  rift-test ~/.pyenv/versions/3.10.7/bin/poetry shell                                                       14.09.22 13:43
Spawning shell within ~/rift-test-lajiwy9G-py3.10
➜  rift-test . ~/rift-test-lajiwy9G-py3.10/bin/activate        14.09.22 13:43
(rift-test-py3.10) ➜  rift-test rift init myproject                                                          14.09.22 13:43
zsh: command not found: rift
(rift-test-py3.10) ➜  rift-test which rift                                                                   14.09.22 13:43
rift not found

```

pyproject.toml file:

```bash

(rift-test-py3.10) ➜  rift-test cat pyproject.toml                                                           14.09.22 13:43
[tool.poetry]
name = "rift-test"
version = "0.1.0"
description = ""
authors = ["Hiyorimi "]
readme = "README.md"
packages = [{include = "rift_test"}]

[tool.poetry.dependencies]
python = "^3.10"
rift-framework = "^0.7.1"


[build-system]
requires = ["poetry-core"]
build-backend = "poetry.core.masonry.api"
```

**Comment by AminRezaei0x443** [comment]
Although these poetry issues will be addressed and resolved, I do not believe that utilizing poetry (or any other package manager) for `rift` end-users and end-projects is a best practice. Because this is related to #7, Ill address this best practice issue separately in here for further reference.

I believe this because `rift` brings new notions that are distinct from a typical `Python` project. To be more specific, we use `Python` in two primary scenarios:
1. `rift` core development: We utilize `Python` to process `Python` contracts and compile them to `FunC`. `rift` is a standard Python package with the usual python code.
2. `rift` contracts: Smart contracts are written in `Python Syntax` and then patched and traced by `rift` to be compiled into `FunC`; this is not a typical `Python` project because behaviors are altered and Python is not targeted. We refer to them as `rift` projects rather than `Python` projects.

So a user would just install `rift` and begin building smart contracts using basic `Python Syntax`, without having to bother about packaging or dependencies. To be more specific, in the future, it is feasible to eliminate the Python dependence and release `rift` as a standalone tool (e.g. by using an internal lightweight python interpreter that is for example a WASM binary). Or even drift away from python syntax and customize it slowly into a new `rift` syntax.

In the future, we may include library and module support, as well as an optional `rift registry`, so that individuals can share basic contract codes and libraries, and developers may utilize them by referencing them rather than copying them into the project.

The main goal is that after installing `rift`, the developer may focus just on `rift` concepts and simple Python syntax, which will be covered in our documentation. `rift` will take care of everything else and the developer will use only the `rift` command-line.

**Comment by Hiyorimi** [comment]
Since #7 works fine from inside the environment, there is still issue with running in poetry:

```bash
➜  testing_rift python3 -m poetry run rift init toad-2-token                                                                                  15.09.22 10:36
Initializing new rift project ...
Traceback (most recent call last):
  File "~/virtualenvs/testing-rift-eisqpOW1-py3.10/bin/rift", line 8, in <module>
    sys.exit(entry())
  File "~/virtualenvs/testing-rift-eisqpOW1-py3.10/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "~/virtualenvs/testing-rift-eisqpOW1-py3.10/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "~/virtualenvs/testing-rift-eisqpOW1-py3.10/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "~/virtualenvs/testing-rift-eisqpOW1-py3.10/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "~/virtualenvs/testing-rift-eisqpOW1-py3.10/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "~/virtualenvs/testing-rift-eisqpOW1-py3.10/lib/python3.10/site-packages/rift/cli/commands/init.py", line 29, in init
    write_contract(base, pjoin(path, name, "contracts", f"{name}.py"))
  File "~/virtualenvs/testing-rift-eisqpOW1-py3.10/lib/python3.10/site-packages/rift/bases/util.py", line 18, in write_contract
    fd = open(dst, "w")
PermissionError: [Errno 13] Permission denied: './toad-2-token/contracts/toad-2-token.py'
```

**Comment by AminRezaei0x443** [comment]
So, would you please mention your OS and its version, so I can try reproducing the issue on my system?

**Comment by Hiyorimi** [comment]
MacOS Big Sur 11.6
Python 3.10.6

**Comment by AminRezaei0x443** [comment]
I guess this issue arises from the `umask` of the process which masks the provided permissions. Would you please try installing from the source and from the `bugfix` branch to see if it fixes the issue?

```
git clone https://github.com/decentralized-builder/rift
cd rift
git checkout bugfix
pip install -e .
rift init test-project
```

**Comment by Hiyorimi** [comment]
```bash
Successfully installed rift-framework-0.7.2
(testing-rift-py3.10) ➜  rift git:(bugfix) rift init test-project                                                                             15.09.22 11:21
Initializing new rift project ...
Error creating project, dir exists!
(testing-rift-py3.10) ➜  rift git:(bugfix) ls                                                                                                 15.09.22 11:21
CODE_OF_CONDUCT.md LICENSE            assets             pyproject.toml     rift               test
CONTRIBUTING.md    README.md          poetry.lock        requirements.txt   setup.py           tox.ini
```

**Comment by AminRezaei0x443** [comment]
Let's check the stacktrace, I should add logs so users can directly upload it to issue easily. for now please:

```
git pull
# should not need this but anyway
pip install -e .
rift init test-project
```

**Comment by AminRezaei0x443** [comment]
Please check it again, that error was from my side and is now fixed in `bugfix`. No need to check the logs if it was for makedirs exception.

**Comment by AminRezaei0x443** [comment]
All poetry and permission-related issues were fixed in favor of #11. Closing as tested successfully on macOS environment. 
Please reopen if any problem persists.
